extensions/positron-notebook-controllers/src/notebookController.ts

```diff
@@ -18,6 +18,21 @@ enum NotebookCellOutputType {
 	ExecuteResult = 'execute_result',
 }
 
+/** An event which fires when a notebook execution starts. */
+export interface DidStartExecutionEvent {
+	/** The notebook cells being executed. */
+	cells: vscode.NotebookCell[];
+}
+
+/** An event which fires when a notebook execution ends. */
+export interface DidEndExecutionEvent {
+	/** The notebook cells that were executed. */
+	cells: vscode.NotebookCell[];
+
+	/** The duration of the execution in milliseconds. */
+	duration: number;
+}
```

Comment:
Would we potentially want to add the execution end type? E.g. `'success' | 'error' | 'canceled'`?